subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/resolver/IvyResolver.java

```diff
@@ -153,6 +162,24 @@ protected MutableIvyModuleResolveMetadata parseMetaDataFromResource(ModuleCompon
         return metaData;
     }
 
+    @Nullable
+    @Override
+    protected MutableIvyModuleResolveMetadata parseMetaDataFromArtifact(ModuleComponentIdentifier moduleComponentIdentifier, ExternalResourceArtifactResolver artifactResolver, ResourceAwareResolveResult result) {
+        MutableIvyModuleResolveMetadata metadata = super.parseMetaDataFromArtifact(moduleComponentIdentifier, artifactResolver, result);
+        if (useGradleMetadata) {
+            LocallyAvailableExternalResource resource = artifactResolver.resolveArtifact(new DefaultModuleComponentArtifactMetadata(moduleComponentIdentifier, new DefaultIvyArtifactName(moduleComponentIdentifier.getModule(), "module", "module")), result);
+            if (resource != null) {
+                // Use default empty metadata when the Ivy file isn't present
+                if (metadata == null) {
+                    ModuleVersionIdentifier mvi = moduleIdentifierFactory.moduleWithVersion(moduleComponentIdentifier.getGroup(), moduleComponentIdentifier.getModule(), moduleComponentIdentifier.getVersion());
+                    metadata = new DefaultMutableIvyModuleResolveMetadata(mvi, moduleComponentIdentifier);
+                }
+                moduleMetadataParser.parse(resource, metadata);
+            }
+        }
+        return metadata;
```

Comment:
:x: This implementation is nearly identical to the one in `MavenResolver`. Should pull this up into superclass.
In fact, I think the current superclass implementation is now unused.